mike_skalandunas/test/test.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+var chai = require('chai');
+var chaiHttp = require('chai-http');
+chai.use(chaiHttp);
+var expect = chai.expect;
+var fs = require('fs');
+var currentTime = require('../lib/date.js');
+
+process.env.MONGOLAB_URI = process.env.MONGOLAB_URI || 'mongodb://localhost/tapir_test';
+require(__dirname + '/../server');
+var mongoose = require('mongoose');
+var Tapir = require(__dirname + '/../models/tapir');
+
+var testTapir = {name: 'floppy',
+  age: 15,
+  weight: 550
+}
+
+describe('API that keeps track of various tapirs', function() {
+  after(function(done) {
+    mongoose.connection.db.dropDatabase(function() {
+      done();
+    });
+  });
+
+  it('should be able to add a new tapir', function(done) {
+    chai.request('localhost:3000')
+      .post('/api/tapirs')
+      .send(testTapir)
+      .end(function(err, res) {
+        expect(err).to.eql(null);
+        expect(res.status).to.eql(200);
+        expect(res.body).to.have.property('_id');
+        done();
+      })
+  })
+
+  it('should log the creation of a tapir, the time, and the date', function(done) {
+    chai.request('localhost:3000')
+      .post('/api/tapirs')
+      .send(testTapir)
+      .end(function(err, res) {
+        expect(err).to.eql(null);
+        expect(res.status).to.eql(200);
+        expect(currentTime()).to.eql(console.log('New tapir added to database at [' + new Date() + '].'));
```

Comment:
Yeah, you'd think so but it surprisingly works.  I tried it on my laptop,
desktop, and an old laptop and they're all working.

On Mon, Nov 16, 2015 at 10:04 AM, Paul Laskowski <notifications@github.com>
wrote:

> In mike_skalandunas/test/test.js
> <https://github.com/codefellows-sea-d45-javascript/mongo_backed_rest_api/pull/2#discussion_r44958556>
> :
>
> > +      .end(function(err, res) {
> > +        expect(err).to.eql(null);
> > +        expect(res.status).to.eql(200);
> > +        expect(res.body).to.have.property('_id');
> > +        done();
> > +      })
> > +  })
> > +
> > +  it('should log the creation of a tapir, the time, and the date', function(done) {
> > +    chai.request('localhost:3000')
> > +      .post('/api/tapirs')
> > +      .send(testTapir)
> > +      .end(function(err, res) {
> > +        expect(err).to.eql(null);
> > +        expect(res.status).to.eql(200);
> > +        expect(currentTime()).to.eql(console.log('New tapir added to database at [' + new Date() + '].'));
>
> does this even work. I would assume that running this test on a slow
> processing computer would fail.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codefellows-sea-d45-javascript/mongo_backed_rest_api/pull/2/files#r44958556>
> .
>



-- 
Mike Skalandunas
203.548.1521
www.mikeskalandunas.com
*Composer & Audio Designer for Video Games*